opentelemetry-sdk/src/logs/record.rs

```diff
@@ -1,10 +1,20 @@
+use super::growable_array::GrowableArray;
 use opentelemetry::{
     logs::{AnyValue, Severity},
     trace::{SpanContext, SpanId, TraceFlags, TraceId},
     Key,
 };
 use std::{borrow::Cow, time::SystemTime};
 
+// According to a Go-specific study mentioned on https://go.dev/blog/slog,
+// up to 5 attributes is the most common case. We have chosen 8 as the default
+// capacity for attributes to avoid reallocation in common scenarios.
+const PREALLOCATED_ATTRIBUTE_CAPACITY: usize = 8;
+
+/// A vector of `Option<(Key, AnyValue)>` with default capacity.
+pub type AttributesGrowableArray =
+    GrowableArray<Option<(Key, AnyValue)>, PREALLOCATED_ATTRIBUTE_CAPACITY>;
```

Comment:
`Option<(Key, AnyValue)>` - Why `Option` is needed?

Reply:
While preallocating an array, we need to store some default value (as in Rust we can't have uninitialised variables). And  `AnyValue` being heterogenous doesn't have a clear default state, so it doesn't implement the `Default` trait. Which makes it difficult for us to create pre-allocated array of `(Key, AnyValue)` tuple. `Option` is internally enum, with `None` as default state, along with allocating the enough room for maximum storage needed for  `(Key, AnyValue)`.

Reply:
We'll need to lookat boxing the complex types to keep enum size under control, now that we are preallocated/stack-allocating them.

Reply:
yes exactly my thoughts. Need to revisit #1913 from size prospective, if not perf.